### ISSUE TITLE
Changing binding value should retrigger validations

### ISSUE THREAD

**Issue by guillaumebaril** [Megabit/Blazorise#4326]
**Describe the bug**
If I change the bind value for a component, it doesn't retrigger the validations

**To Reproduce**
If I start typing in the Name component, the ValidationSuccess message is displayed. If I click on the "Change name" button, the name is set to null, and the validation should display the "Enter valid name!" message, but it doesn't.

Use the code below : 

```razor
@page "/"

<Validation Validator="ValidationRule.IsNotEmpty">
    <TextEdit Placeholder="Enter name" @bind-Text="name">
        <Feedback>
            <ValidationNone>Please enter the name.</ValidationNone>
            <ValidationSuccess>Name is good.</ValidationSuccess>
            <ValidationError>Enter valid name!</ValidationError>
        </Feedback>
    </TextEdit>
</Validation>
<Button Clicked="UpdateName">Change name</Button>

@code {
    private string name = null;
    private void UpdateName()
    {
        name = null;
    }
}
```

**Expected behavior**
When we click on the "Change name" button, the validation should retrigger and display the "validation error" message.

**Screenshots**
![image](https://user-images.githubusercontent.com/906761/203632875-251ab0ef-446e-446d-a2af-bc55cfd42739.png)

**Comment by stsrki** [comment]
We already had similar reports in the past. Considering it would probably need to cover significant internal changes and that we need to cover a lot of tests, I think it is best to fix it in 1.2.

A quick try to fix it seems promising:

```diff
public override async Task SetParametersAsync( ParameterView parameters )
{
+    if ( Rendered )
+    {
+        if ( parameters.TryGetValue<string>( nameof( Text ), out var paramText ) && paramText != Text )
+        {
+            ExecuteAfterRender( Revalidate );
+        }
+    }

    await base.SetParametersAsync( parameters );

    if ( ParentValidation != null )
    {
        if ( parameters.TryGetValue<Expression<Func<string>>>( nameof( TextExpression ), out var expression ) )
            await ParentValidation.InitializeInputExpression( expression );

        if ( parameters.TryGetValue<string>( nameof( Pattern ), out var pattern ) )
        {
            // make sure we get the newest value
            var value = parameters.TryGetValue<string>( nameof( Text ), out var paramText )
                ? paramText
                : InternalValue;

            await ParentValidation.InitializeInputPattern( pattern, value );
        }

        await InitializeValidation();
    }
}
```

But it might brake something else in the component lifecycle.

**Comment by guillaumebaril** [comment]
Allright ! I'll follow the update, and thanks again for the great work !
